ansible/idr-searchengine.yml

```diff
@@ -58,82 +60,161 @@
       group: root
       mode: 0755
 
+  - name: Create elasticsearch main nodes directories
+    become: yes
+    file:
+      path: "{{ apps_folder }}/searchengine/elasticsearch/node{{ item }}"
+      state: directory
+      # User id in elasticsearch Docker image
+      owner: 1000
+      group: root
+      mode: 0755
+    with_sequence: start=1 count={{ elasticsearch_no_nodes }}
+
   - name: Create elasticsearch logs directory
     become: yes
     file:
-      path: "{{ apps_folder }}/searchengine/elasticsearch/logs"
+      path: "{{ apps_folder }}/searchengine/elasticsearch/node{{ item }}/logs"
       state: directory
       # User id in elasticsearch Docker image
       owner: 1000
       group: root
       mode: 0755
+    with_sequence: start=1 count={{ elasticsearch_no_nodes }}
 
   - name: Create elasticsearch data directory
     become: yes
     file:
-      path: "{{ apps_folder }}/searchengine/elasticsearch{{ apps_folder }}"
+      path: "{{ apps_folder }}/searchengine/elasticsearch/node{{ item }}/data"
       state: directory
       # User id in elasticsearch Docker image
       owner: 1000
       group: root
       mode: 0755
+    with_sequence: start=1 count={{ elasticsearch_no_nodes }}
+
+  - name: Create ELasticsearch backup folder
+    become: yes
+    file:
+      path: "{{ elasticsearch_backup_folder }}"
+      recurse: yes
+      state: directory
+      owner: 1000
+      group: root
+      mode: 0755
+
+  #Add all elasticsearch nodes
+  - name: Add elastic nodes to elasticsearch_nodes
+    set_fact:
+      elasticsearch_nodes: '{{ elasticsearch_nodes + ["searchengine_elasticsearch_node"+item] }}'
+    with_sequence: start=1 count={{ elasticsearch_no_nodes }}
 
   - name: Create docker network
     become: yes
     docker_network:
       name: searchengine-net
       ipam_config:
         - subnet=10.11.0.0/16
-      #networks_cli_compatible: yes
-      #state: present
 
-  #- name: Pull search engine docker image
-  #  docker_image:
-  #      name: "{{ search_engineicsearch_docker_image }}"
-  #      tag: latest
-  #      source: pull
+  # I got some memory exceptions when start using elasticsearch cluster and
+  # increasing the mmap counts limits fix this issue
+  #https://www.elastic.co/guide/en/elasticsearch/reference/7.16 /vm-max-map-count.html
+  - name: set vm.max_map_count to 262144 in sysctl
```

Comment:
I needed to set `become: yes` here to avoid a permission issue on deployment

Reply:
On `test112` during the initial deployment of this playbook. Possibly this is unnecessary if the variable is already set?